starter-code/src/viking.js

```diff
@@ -1,11 +1,114 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function() {
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.battleCry = function() {
+  return "Odin Owns You All!";
+};
+
+Viking.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+  return this.health > 0 ? this.name + " has received " + damage + " points of damage" : this.name + " has died in act of combat";
+};
```

Comment:
Con esto, estaríais machacando la función original que pertenece al prototype de Soldier. Para hacerlo correctamente deberíais modificar la original, no pisarla; eso se hace haciendo un call de dicha función dentro del `receiveDamage` de Viking -> `Soldier.prototype.receiveDamage.call(this, damage)` dentro de `Viking.prototype.receiveDamage(damage)`